certbot-ci/certbot_integration_tests/certbot_tests/test_main.py

```diff
@@ -23,6 +24,72 @@ def context(request):
         integration_test_context.cleanup()
 
 
+def test_basic_commands(context):
+    """Test simple commands on Certbot CLI."""
+    # TMPDIR env variable is set to workspace for the certbot subprocess.
+    # So tempdir module will create any temporary files/dirs in workspace,
+    # and its content can be tested to check correct certbot cleanup.
+    initial_count_tmpfiles = len(os.listdir(context.workspace))
+
+    context.certbot(['--help'])
+    context.certbot(['--help', 'all'])
+    context.certbot(['--version'])
+
+    with pytest.raises(subprocess.CalledProcessError):
+        context.certbot(['--csr'])
+
+    new_count_tmpfiles = len(os.listdir(context.workspace))
+    assert initial_count_tmpfiles == new_count_tmpfiles
+
+
+def test_hook_dirs_creation(context):
+    """Test all hooks directory are created during Certbot startup."""
+    context.certbot(['register'])
+
+    for hook_dir in misc.list_renewal_hooks_dirs(context.config_dir):
+        assert os.path.isdir(hook_dir)
+
+
+def test_registration_override(context):
+    """Test correct register/unregister, and registration override."""
+    context.certbot(['register'])
+    context.certbot(['unregister'])
+    context.certbot(['register', '--email', 'ex1@domain.org,ex2@domain.org'])
+
+    # TODO: When `certbot register --update-registration` is fully deprecated,
+    #  delete the two following deprecated uses
+    context.certbot(['register', '--update-registration', '--email', 'ex1@domain.org'])
+    context.certbot(['register', '--update-registration', '--email', 'ex1@domain.org,ex2@domain.org'])
+
+    context.certbot(['update_account', '--email', 'example@domain.org'])
+    context.certbot(['update_account', '--email', 'ex1@domain.org,ex2@domain.org'])
+
+
+def test_prepare_plugins(context):
+    """Test that plugins are correctly instantiated and displayed."""
+    output = context.certbot(['plugins', '--init', '--prepare'])
+
+    assert 'webroot' in output
+
+
+def test_http_01(context):
+    """Test the HTTP-01 challenge using standalone plugin."""
+    # We start a server listening on the port for the
+    # TLS-SNI challenge to prevent regressions in #3601.
+    with misc.create_http_server(context.tls_alpn_01_port):
+        certname = context.get_domain('le2')
+        context.certbot([
+            '--domains', certname, '--preferred-challenges', 'http-01', 'run',
+            '--cert-name', certname,
+            '--pre-hook', 'echo wtf.pre >> "{0}"'.format(context.hook_probe),
+            '--post-hook', 'echo wtf.post >> "{0}"'.format(context.hook_probe),
+            '--deploy-hook', 'echo deploy >> "{0}"'.format(context.hook_probe)
+        ])
+
+    assert_hook_execution(context.hook_probe, 'deploy')
+    assert_save_renew_hook(context.config_dir, certname)
```

Comment:
I suspect we just want to change the code in `assert_save_renew_hook`, but this isn't testing the same thing as [CheckSavedRenewHook](https://github.com/certbot/certbot/blob/d9880721b31ab4e9e9d2fdc38d83ff2bd9078378/tests/certbot-boulder-integration.sh#L142). The shell code is trying to check that there is an entry for `renew_hook` saved into the renewal configuration file.

Reply:
Indeed, I did not interpretate this assertion correctly. This is fixed.